lib/elixir/lib/calendar/duration.ex

```diff
@@ -331,4 +333,107 @@ defmodule Duration do
         raise ArgumentError, ~s/failed to parse duration "#{string}". reason: #{inspect(reason)}/
     end
   end
+
+  @doc """
+  Converts the given duration to [ISO 8601-2:2019](https://en.wikipedia.org/wiki/ISO_8601) format.
+
+  Note this function implements the *extension* of ISO 8601:2019. This extensions allows weeks to
+  appear in any case, which is not the case in the base standard. `"P3M3W"` is not allowed in
+  ISO 8601-1, but it is in ISO 8601-2. Thus, any duration can be output by this function.
+
+  ## Examples
+
+      iex> Duration.new!(year: 3) |> Duration.to_iso8601()
+      "P3Y"
+      iex> Duration.new!(day: 28, hour: 6, minute: 42, second: 12) |> Duration.to_iso8601()
+      "P28DT6H42M12S"
+      iex> Duration.new!(second: 30) |> Duration.to_iso8601()
+      "PT30S"
+      iex> Duration.new!(second: 30) |> Duration.multiply(-1) |> Duration.to_iso8601()
+      "PT-30S"
+      iex> Duration.new!([]) |> Duration.to_iso8601()
+      "PT0S"
+
+      iex> Duration.new!(hour: 2, microsecond: {2, 6}) |> Duration.to_iso8601()
+      "PT2H0.000002S"
+      iex> Duration.new!(hour: 2, microsecond: {2, 3}) |> Duration.to_iso8601()
+      "PT2H0.000S"
+      iex> Duration.new!(hour: 2, microsecond: {2, 0}) |> Duration.to_iso8601()
+      "PT2H0S"
+      iex> Duration.new!(microsecond: {2, 0}) |> Duration.to_iso8601()
+      "PT0S"
+      iex> Duration.new!(microsecond: {2_000_000, 6}) |> Duration.to_iso8601()
+      "PT2.000000S"
+      iex> %Duration{second: 1, microsecond: {-1_200_000, 4}} |> Duration.to_iso8601()
+      "PT-0.2000S"
```

Comment:
Those look more like unit tests, shall we move them to the duration_test.exs file?

Reply:
The test coverage looks good to me!